src/main/java/seedu/address/logic/commands/AddTagCommand.java

```diff
@@ -0,0 +1,133 @@
+            + "the tag is valid and is not already tagged to the contact (case-insensitive).\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_TAG + "TAG]\n"
+            + "Example: " + COMMAND_WORD + " 2 "
+            + PREFIX_TAG + "friends";
+
+    public static final String MESSAGE_ADD_TAG_SUCCESS = "Added tag: %1$s";
+    public static final String MESSAGE_DUPLICATE_TAG = "This person is already tagged to %1$s.";
+
+    private final Index index;
+    private final Tag tag;
+
+    /**
+     * Constructs an {@code AddTagCommand} with the given {@code Index} and {@code Tag}.
+     *
+     * @param index of the person in the filtered person list to add the {@code Tag}.
+     * @param tag to be tagged to the {@code Person} specified by {@code index}.
+     */
+    public AddTagCommand(Index index, Tag tag) {
+        requireNonNull(index);
+        requireNonNull(tag);
+
+        this.index = index;
+        this.tag = tag;
+    }
+
+    /**
+     * Adds a tag to an existing person in the address book, if the tag does not exist.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return the command result after the command execution.
+     * @throws CommandException if the {@code Tag} already exists or the {@code Index} is invalid.
+     */
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        if (personToEdit.hasTag(this.tag)) {
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_TAG, this.tag));
+        }
+
+        Person editedPerson = createPersonWithAddedTag(personToEdit, this.tag);
+
+        model.setPerson(personToEdit, editedPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(String.format(MESSAGE_ADD_TAG_SUCCESS, editedPerson));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of personToEdit {@code Person}
+     * added with tagToAdd {@code Tag}.
+     */
+    private static Person createPersonWithAddedTag(Person personToEdit, Tag tagToAdd) {
+        assert personToEdit != null;
+
+        Name updatedName = personToEdit.getName();
+        Phone updatedPhone = personToEdit.getPhone();
+        Email updatedEmail = personToEdit.getEmail();
+        Address updatedAddress = personToEdit.getAddress();
+        Set<Tag> updatedTags = addTagToSet(personToEdit.getTags(), tagToAdd);
+
+        return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedTags);
```

Comment:
There will be an error when both our commits merge in, but no worries, I'll pull and commit to fix it after the merges.